### PR TITLE
chore: log relay errors only once per app

### PR DESCRIPTION
**Pull request by im-adithya**
Now we only log once instead of per subscription which bloats DD

### CODE REVIEW

File: internal/nostr/nostr.go [hunk 1]
```diff
@@ -728,28 +727,13 @@ func (svc *Service) startSubscription(ctx context.Context, subscription *Subscri
...
-			svc.Logger.WithError(err).WithFields(logrus.Fields{
-				"subscription_id": subscription.Uuid,
-				"relay_url":       subscription.RelayUrl,
-			}).Errorf("Failed to connect to relay, retrying in %vs...", waitToReconnectSeconds)
 			continue
 		}
 
 		sub, err := relay.Subscribe(ctx, []nostr.Filter{*filter})
 		if err != nil {
-			// TODO: notify user about subscription failure
```
Comment (rolznz) [inline internal/nostr/nostr.go:746]: I think we should still log an error here, shouldn't we?

File: internal/nostr/nostr.go [hunk 2]
```diff
@@ -713,7 +713,6 @@ func (svc *Service) startSubscription(ctx context.Context, subscription *Subscri
 	var relay *nostr.Relay
 	var isCustomRelay bool
 	var err error
-	waitToReconnectSeconds := 0
```
Comment (rolznz) [inline internal/nostr/nostr.go:716]: A few lines down (line 726) we cancel the subscription and return if there is an error. I think we should only exit this loop when the `context.Done` channel receives a message, and we should handle it in one place like we do in Alby Hub.

File: internal/nostr/nostr.go [hunk 3]
```diff
@@ -917,14 +899,43 @@ func (svc *Service) getRelayConnection(ctx context.Context, customRelayURL strin
...
-		relay, err := nostr.RelayConnect(svc.Ctx, svc.Cfg.DefaultRelayURL)
+		relay, err := svc.relayConnectWithBackoff(svc.Ctx, svc.Cfg.DefaultRelayURL)
 		if err == nil {
 			svc.Relay = relay
 		}
 		return svc.Relay, false, err
 	}
 }
 
+func (svc *Service) relayConnectWithBackoff(ctx context.Context, relayURL string) (relay *nostr.Relay, err error) {
```
Comment (rolznz) [inline internal/nostr/nostr.go:910]: Nice idea :+1:

File: internal/nostr/nostr.go (hunk 1 above)
Comment (im-adithya) [reply internal/nostr/nostr.go:746]: This error never happened (in absolute terms), that's why I was confident of it, but doesn't hurt to log, should Ii add it back?

**Comment by im-adithya** [comment]
The feedback here is noted, and will be addressed in a PR which aims to separate one-off subscriptions (as both comments refer to code in `startSubscription` function

**Pull request merged**